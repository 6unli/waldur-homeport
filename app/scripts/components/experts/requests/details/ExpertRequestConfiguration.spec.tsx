import { render } from 'enzyme';
import * as React from 'react';

import { ExpertRequestConfiguration } from './ExpertRequestConfiguration';

const renderComponent = model => {
  const config = {
    order: ['milestones'],
    options: {
      milestones: {
        type: 'html_text',
        label: 'Milestones',
      },
    },
  };
  const wrapper = render(
    <ExpertRequestConfiguration
      model={model}
      config={config}
    />
  );
  return wrapper.html();
};

describe('ExpertRequestConfiguration', () => {
  it('renders HTML field as is', () => {
    const html = renderComponent({
      milestones: '<ul><li>Deploy virtual machine</li></ul>',
    });
    expect(html).toMatchSnapshot();
  });

  it('renders placeholder if value if missing', () => {
    const html = renderComponent({milestones: undefined});
    expect(html).toMatchSnapshot();
  });
});
