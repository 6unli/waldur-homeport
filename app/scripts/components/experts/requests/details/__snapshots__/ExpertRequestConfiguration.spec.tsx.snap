// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpertRequestConfiguration renders HTML field as is 1`] = `"<div class=\\"m-b-xs\\"><dt>Milestones</dt><dd><div><ul><li>Deploy virtual machine</li></ul></div></dd></div>"`;

exports[`ExpertRequestConfiguration renders placeholder if value if missing 1`] = `"<div class=\\"m-b-xs\\"><dt>Milestones</dt><dd>N/A</dd></div>"`;
